core/contracts/financial-templates/implementation/demo/DepositBox.sol

```diff
@@ -0,0 +1,346 @@
+ * @notice This is a minimal example of a financial template that depends on price requests from the DVM.
+ * Specifically, this contract should be thought of as a "Deposit Box" into which the user deposits some ERC20 collateral.
+ * The main feature of this box is that the user can withdraw their ERC20 corresponding to a desired USD amount.
+ * When the user wants to make a withdrawal, a price request is enqueued with the UMA DVM.
+ * For simplicty, the user is constrained to have one outstanding withdrawal request at any given time.
+ * Regular fees are charged on the collateral in the deposit box throughout
+ * the lifetime of the deposit box, and final fees are charged on each price request.
+ *
+ * This example is intended to accompany a technical tutorial for how to integrate the DVM into a project.
+ * The main feature this demo serves to showcase is how to build a financial product on-chain that "pulls" price requests
+ * from the DVM on-demand, which is an implementation of the "priceless" oracle framework.
+ *
+ * The typical user flow would be:
+ * - User sets up a deposit box for the (wETH - USD) price-identifier. The "collateral currency" in this deposit box is therefore wETH.
+ *   The user can subsequently make withdrawal requests for USD-denominated amounts of wETH.
+ * - User deposits 10 wETH into their deposit box.
+ * - User later requests to withdraw $100 USD of wETH.
+ * - DepositBox asks DVM for latest wETH/USD exchange rate.
+ * - DVM resolves the exchange rate at: 1 wETH is worth 200 USD.
+ * - DepositBox transfers 0.5 wETH to user.
+ */
+contract DepositBox is FeePayer, AdministrateeInterface, ContractCreator {
+    using SafeMath for uint256;
+    using FixedPoint for FixedPoint.Unsigned;
+    using SafeERC20 for IERC20;
+
+    // Represents a single caller's deposit box. All collateral is held by this contract.
+    struct DepositBoxData {
+        // Requested amount of collateral, denominated in quote asset of the price identifier.
+        // Example: If the price identifier is wETH-USD, and the `withdrawalRequestAmount = 100`, then
+        // this represents a withdrawal request for 100 USD worth of wETH.
+        FixedPoint.Unsigned withdrawalRequestAmount;
+        // Timestamp of the latest withdrawal request. A withdrawal request is pending if `requestPassTimestamp != 0`.
+        uint256 requestPassTimestamp;
+        // Raw collateral value. This value should never be accessed directly -- always use _getFeeAdjustedCollateral().
+        // To add or remove collateral, use _addCollateral() and _removeCollateral().
+        FixedPoint.Unsigned rawCollateral;
+    }
+
+    // Maps addresses to their deposit boxes. Each address can have only one position.
+    mapping(address => DepositBoxData) public depositBoxes;
+
+    // Unique identifier for DVM price feed ticker.
+    bytes32 public priceIdentifier;
+
+    // Similar to the rawCollateral in DepositBoxData, this value should not be used directly.
```

Comment:
as they should not be accessed directly do you want to make it private?

Reply:
yeah I'll do that. I made it public out of consistency with the PPM, any reason we made it public there?